src/bin.ts

```diff
@@ -50,8 +52,20 @@ export function main() {
 }
 
 function isTsEsmLoaderNeeded(cmd: string, args: string[]) {
-  // HACK: Ugly way to check if it's ts-node instead of node
-  return [cmd, ...args].includes("ts-node");
+  // HACK: Ugly way to check
+  const tsConfigTsNodeEsmIsTrue = () => {
+    const tsConfigPath = join(cwd(), "tsconfig.json");
+    if (!existsSync(tsConfigPath)) return false;
+
+    const tsConfigContent = readFileSync(tsConfigPath, "utf-8");
+    const tsConfig = parseConfigFileTextToJson("", tsConfigContent).config as {
+      "ts-node"?: { esm?: boolean };
```

Comment:
This will cause an error when `tsconfig.json` is malformed. You should cast it to `unknown` and check the types explicitly instead.